engine/runtime/src/main/scala/org/enso/interpreter/runtime/util/DiagnosticFormatter.scala

```diff
@@ -60,88 +90,142 @@ class DiagnosticFormatter(
     case None => false
   }
 
-  private def textAndLocation(): (String, SourceSection) = {
+  sealed protected trait Location {
```

Comment:
`Location`? We already have a `Location` class elsewhere. Please choose a different name.